packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -240,6 +240,8 @@ export default class AllureReporter extends WDIOReporter {
 
         if (cid) {
             this._state.currentTest.addLabel(LabelName.THREAD, cid)
+            this._state.currentTest.historyId = cid
```

Comment:
You cannot use `cid` as the `historyId` because the `historyId` must remain the same for each test run. Actually, it is a unique identifier for the specific test between all runs of specific test.
`cid` is a random value.

I'd suggest to use `md5(suiteName+testCaseName)` as  `historyId`.